torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -229,25 +229,27 @@ Value* CloneValueFromListConstruct(
   if (elem->cast<IntType>()) {
     scalar_type = at::kLong;
 
-    auto lc_node = v->node();
-    // ListConstruct Int[] output case, we need to transform to ONNX
-    // Concat to ensure the output is a single tensor(dynamic) type in
-    // order to be consumed as inputs
-    std::vector<Value*> unsqueezed;
-    for (auto* input : lc_node->inputs()) {
-      auto new_input = n_graph->addInput();
-      new_input->copyMetadata(input);
-      Node* unsqueezed_node = createONNXUnsqueeze(
-          n_graph.get(), n_graph->return_node(), new_input, 0, opset_version);
-      unsqueezed.emplace_back(unsqueezed_node->output());
-    }
-    Node* concat_node =
-        n_graph->insertNode(n_graph->create(::c10::onnx::Concat, 1));
-    concat_node->i_(attr::axis, 0);
-    for (auto v : unsqueezed) {
-      concat_node->addInput(v);
+    if (!v->node()->inputs().empty()) {
```

Comment:
Interesting. Is this part of the code duplicate of the peephole pass?
Do we want to keep both of these?

Reply:
Yes. This similar code block exists in both shape_type_inference and onnx_peephole pass.

1. If we keep this only in shape_type_inference, the switch _onnx_shape_inference would be a little tricky. Because in some cases, we might set it to False to stop shape inference work.

2. If we keep this only in onnx/peephole.cpp file, I checked the tests work as expected. But according to the comments in shape_type_inference file below, seems like we want to retire it one day. @BowenBao  Do you have an idea about why we put the same code in shape_type_inference.cpp too?

  // In jit/passes/onnx/peephole.cpp::eraseListConstruct,
  // prim::ListConstruct is converted to onnx::Concat. The conversion should
  // eventually be moved to symbolic. For now, treat this operator as
  // special case, and change from list type to tensor type. The scalar type
  // is preserved. If the elemtype is Int, insert a onnx::Concat node into
  // the graph.

Reply:
consider abstracting the same logic into a function for reuse. For current design this cannot be retired yet. conversion of prim::ListConstruct depends on later nodes that consume it. So we cannot convert it in symbolic function upfront. Thus here in shape inference, we still observe prim::ListConstruct as upstream input, thus require special handle.